### PR TITLE
feat(rust): disable `FlowControl` for loopback tcp connections and listeners

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/util.rs [hunk 1]
```diff
@@ -54,56 +54,75 @@ pub async fn multiaddr_to_route(
...
-        flow_controls,
-        &flow_control_id,
-    ));
+    let mut flow_control_id = None;
+    let mut number_of_tcp_hops = 0;
 
     while let Some(p) = it.next() {
         match p.code() {
             Ip4::CODE => {
+                if number_of_tcp_hops >= 1 {
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_api/src/util.rs:64]: This condition could be lifted out of the `match`

**Review by etorreborre — APPROVED** [review]
👍

File: implementations/rust/ockam/ockam_api/src/util.rs (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_api/src/util.rs:64]: Not sure

Comment (davide-baldo) [reply implementations/rust/ockam/ockam_api/src/util.rs:64]: We want the condition only for dns,ipv4 and ipv6

Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/util.rs:64]: Sorry I skipped the other cases with code folding. I wish we had some common abstraction for those 3 cases but maybe there's nothing that really fits.

**Pull request merged**